torchdynamo/variables/functions.py

```diff
@@ -325,3 +325,121 @@ def reconstruct(self, codegen):
+        self.fn = fn
+
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        import random
+        import string
+
+        import torchdynamo.config as config
+        import torchdynamo.mutation_guard as mutation_guard
+        from torchdynamo.eval_frame import export
+        from torchdynamo.mutation_guard import GenerationTracker
+        from torchdynamo.source import LocalSource
+        from torchdynamo.source import NNModuleSource
+
+        true_fn = args[1].get_function()
+        false_fn = args[2].get_function()
+
+        true_func_args_packed = args[3]
+        false_func_args_packed = args[4]
+
+        true_func_args = []
+        false_func_args = []
+
+        for item in true_func_args_packed.items:
+            assert isinstance(
+                item, variables.TensorVariable
+            ), "Conditionals must take only tensor variables"
+            true_func_args.append(item.proxy.node.meta["example_value"])
+
+        for item in false_func_args_packed.items:
+            assert isinstance(
+                item, variables.TensorVariable
+            ), "Conditionals must take only tensor variables"
+            false_func_args.append(item.proxy.node.meta["example_value"])
+
+        assert (
+            config.fake_tensor_propagation is False
+        ), "Fake Tensor Propogation must be disabled for conditional capture"
+        out_true = export(true_fn, *true_func_args)
+        out_true_graph = out_true[0]
+
+        # `export` causes generation increments, and this means that the second exported graph
+        # is incorrectly tagged as "dynamic". This is a disgusting hack, and we can do a little
+        # better job here probably with another type of export that avoids generational incrementing
+        # via supplying a different function on the context, rather than this.
+        mutation_guard.OVERRIDE_GENERATION_TAGGING = 0
+        out_false = export(false_fn, *false_func_args)
+        mutation_guard.OVERRIDE_GENERATION_TAGGING = None
+
+        # Adjust the generation back to what it should be
+        GenerationTracker.generation += 1
+
+        out_false_graph = out_false[0]
+
+        def rand_slug():
+            return "".join(random.choices(string.ascii_uppercase + string.digits, k=8))
+
+        true_name = "true_graph_" + rand_slug()
+        false_name = "false_graph_" + rand_slug()
+
+        true_source = LocalSource(out_true_graph)
+        true_source.local_name = true_name
+        false_source = LocalSource(out_false_graph)
+        false_source.local_name = false_name
+
+        tx.output.add_submodule(
+            out_true_graph, true_name, source=NNModuleSource(true_source)
+        )
+        tx.output.add_submodule(
+            out_false_graph, false_name, source=NNModuleSource(false_source)
+        )
+
+        true_graph_node = tx.output.create_proxy(
+            "get_attr",
```

Comment:
Why is this get_attr instead of call_module? Aren't true and false branch just submodules on the output graph